samples/core/blur/blur.cpp

```diff
@@ -499,8 +513,28 @@ std::tuple<bool, bool, bool> BlurCppExample::query_capabilities()
     bool use_subgroup_exchange_relative = cl::util::supports_extension(
         device, "cl_khr_subgroup_shuffle_relative");
 
+    // 5) Query OpenCL version to compile for.
+    // If no -cl-std option is specified then the highest 1.x version
+    // supported by each device is used to compile the program. Therefore,
+    // it's only necessary to add the -cl-std option for 2.0 and 3.0 OpenCL
+    // versions.
+    const std::string dev_version = device.getInfo<CL_DEVICE_VERSION>();
+    cl::string compiler_options;
+    constexpr int max_major_version = 3;
```

Comment:
I suspect other samples will need to do something similar, so this would be a great SDK utility function.  This code from the OpenCL CTS may be a useful starting point:

https://github.com/KhronosGroup/OpenCL-CTS/blob/cd74e0264391e0424a27f657852fd1e9f10bf42e/test_common/harness/kernelHelpers.cpp#L1645

Strictly speaking, this really ought to be checking for the OpenCL C versions supported by the device, not for the OpenCL version supported by the device, but given the current couplings between the OpenCL version and the OpenCL C versions I _think_ this is OK.